src/lib/core/cbus.h

```diff
@@ -466,6 +488,66 @@ cbus_unpair(struct cpipe *dest_pipe, struct cpipe *src_pipe,
 	    void (*unpair_cb)(void *), void *unpair_arg,
 	    void (*process_cb)(struct cbus_endpoint *));
 
+/**
+ * A  uni-directional FIFO queue from any thread to cord.
+ * It is a light version of the cpipe, since some features are not needed for
+ * 3td party threads or they cannot be implemented.
+ *
+ */
+struct lcpipe {
+	/** Staging area for pushed messages */
+	struct stailq input;
+	/** Counters are useful for finer-grained scheduling. */
+	int n_input;
+	/**
+	 * When pushing messages, keep the staged input size under
+	 * this limit (speeds up message delivery and reduces
+	 * latency, while still keeping the bus mutex cold enough).
+	 */
+	int max_input;
+	/**
+	 * The cbus endpoint at the destination cord to handle
+	 * flushed messages.
+	 */
+	struct cbus_endpoint *endpoint;
+};
+
+/**
+ * Create and initialize a pipe and connect it to the consumer.
+ * The call returns only when the consumer, identified by consumer name, has
+ * joined the bus.
+ **/
+struct lcpipe *
+lcpipe_new(const char *consumer);
+
+/**
+ * Destroy a pipe with freeing up occupied memory.
+ * @param pipe lcpipe must have been allocated with a call to `lcpipe_new`.
+ **/
+void
+lcpipe_delete(struct lcpipe *pipe);
+
+/**
+ * Flush all staged messages into the pipe and eventually to the
+ * consumer.
+ */
+void
+lcpipe_flush_input(struct lcpipe *pipe);
```

Comment:
This is questionable whether we should expose those lcpipe and cpipe methods directly with these names instead of using `tnt_` or `box_` or some other prefix common for Tarantool exports. But I will leave that to the later core team reviewers who so gently gave me the place of the first reviewer. At first I suggest to eliminate the duplicated code.

Reply:
Yes, you are right, maybe we should make this in a separate PR? Looks like this not about lcpipe, but about just cpipe refactoring